spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/autoconfig/TraceAutoConfiguration.java

```diff
@@ -70,6 +71,12 @@ public SleuthMetricProperties sleuthMetricProperties() {
 		return new SleuthMetricProperties();
 	}
 
+	@Bean
+	@ConditionalOnMissingBean
```

Comment:
this is why I don't understand the defensive lookup below.. if this allows a default, why would we guard on TraceKeys not found?